src/integritysetup/integritysetup-generator.c

```diff
@@ -0,0 +1,224 @@
+
+                if (len > 0) {
+                        *options_formatted = buff;
+                }
+                return 0;
+        }
+        return -EINVAL;
+}
+
+static int create_disk(
+                const char *name,
+                const char *device,
+                char *options) {
+
+        _cleanup_free_ char *n = NULL, *dd = NULL, *hd = NULL, *e = NULL, *to = NULL, *dev_require = NULL,
+                            *d_escaped = NULL, *hu_escaped = NULL, *name_escaped = NULL, *options_convert = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        int r;
+        char *dmname = NULL;
+
+        assert(name);
+        assert(device);
+
+        name_escaped = specifier_escape(name);
+        if (!name_escaped)
+                return log_oom();
+
+        e = unit_name_escape(name);
+        if (!e)
+                return log_oom();
+
+        r = unit_name_build("systemd-integritysetup", e, ".service", &n);
+        if (r < 0)
+                return log_error_errno(r, "Failed to generate unit name: %m");
+
+        r = unit_name_from_path(device, ".device", &dd);
+        if (r < 0)
+                return log_error_errno(r, "Failed to generate unit name: %m");
+
+        r = generator_open_unit_file(arg_dest, NULL, n, &f);
+        if (r < 0)
+                return r;
+
+        if (options) {
+                r = options_string(options, &options_convert);
+                if (r < 0)
+                        return r;
+        }
+
+        fprintf(f,
+                "[Unit]\n"
+                "Description=Integrity Setup for %%I\n"
+                "Documentation=man:integritytab(5) man:systemd-integritysetup-generator(8) man:systemd-integritysetup@.service(8)\n"
+                "SourcePath=%s\n"
+                "DefaultDependencies=no\n"
+                "IgnoreOnIsolate=true\n"
+                "After=systemd-udevd-kernel.socket\n"
+                "Before=blockdev@dev-mapper-%%i.target\n"
+                "Wants=blockdev@dev-mapper-%%i.target\n"
+                "Conflicts=umount.target\n"
+                "Before=integritysetup.target\n"
+                "BindsTo=%s\n"
+                "After=%s\n"
+                "Before=umount.target\n",
+                arg_integritytab,
+                dd, dd);
+
+        fprintf(f,
+                "\n"
+                "[Service]\n"
+                "Type=oneshot\n"
+                "RemainAfterExit=yes\n"
+                "TimeoutSec=0\n"
+                "ExecStart=%s open '%s' '%s' %s\n"
+                "ExecStop=%s close '%s'\n",
+                arg_integritysetupbin, device, name_escaped, (options_convert) ? options_convert: "",
+                arg_integritysetupbin, name_escaped);
+
+        r = fflush_and_check(f);
+        if (r < 0)
+                return log_error_errno(r, "Failed to write unit file %s: %m", n);
+
+        r =  generator_add_symlink(arg_dest, "integritysetup.target", "requires", n);
```

Comment:
double space

Reply:
Wow, watch a catch.  I had to stare at this one to see it :-), done